src/rulesets/__tests__/reader.test.ts

```diff
@@ -47,13 +48,53 @@ describe('Rulesets reader', () => {
     });
   });
 
-  it('should inherit properties of extended rulesets', () => {
-    return expect(readRulesFromRulesets(extendsOas2Ruleset)).resolves.toEqual({
+  it('should inherit properties of extended rulesets', async () => {
+    const rules = await readRulesFromRulesets(extendsOas2Ruleset);
+
+    // we pick up recommended rules only from spectral:oas
```

Comment:
Could you elaborate a bit more re what this test is checking?

We might have missed a nuance in the original implementation description. The initial ruleset extends should probably determine the whole thing. For example:

If `base-rulset2.json` has `extends: ['base-ruleset1.json', 'all']`, and `my-ruleset.json` has `extends: ['base-ruleset2.json', 'off']`, running spectral with the `my-ruleset.json` should result all rules being off (from all the extended rulesets). Otherwise it would be pretty confusing to ask for recommended rules and not get them, or to ask for extended rules to start off as off, and not have that be the case, etc.

Thoughts?

Reply:
I think we might have missed it. The current implementation applies the ruleset severity only to direct dependencies (in other words it's one level deep), therefore it won't override the ruleset severity declared in extended rulesets.
I fully agree with you this is very likely to cause confusion. I'll adjust the implementation.